### PR TITLE
Update Auth examples with GoogleSignIn 6.0.0

### LINKED ISSUE #1218: Authentication example should be updated for GoogleSignIn 6.0.0
There are multiple build failures related to the removal of the `GIDSignInDelegate` protocol, `sharedInstance` now a class property, and other changes from https://github.com/google/GoogleSignIn-iOS/blob/main/CHANGELOG.md.

In the meantime a workaround is to specify   `pod 'GoogleSignIn', '~> 5.0'` in the Podfile.

### PR DESCRIPTION
**Pull request by maksymmalyhin**
Fixes #1218.

### CODE REVIEW

**Comment by paulb777** [comment]
This is strange - https://github.com/firebase/quickstart-ios/blob/master/.github/workflows/authentication.yml#L12

**Comment by maksymmalyhin** [comment]
> This is strange - https://github.com/firebase/quickstart-ios/blob/master/.github/workflows/authentication.yml#L12

Yeah, it seems the branch restriction should be removed for pull requests. @morganchen12 Is there any particular reason it was added there?

File: authentication/LegacyAuthQuickstart/AuthenticationExample/AppDelegate.m [hunk 1]
```diff
@@ -35,11 +35,6 @@ - (BOOL)application:(UIApplication *)application
   [FIRApp configure];
   // [END firebase_configure]
 
-  // [START setup_gidsignin]
```
Comment (maksymmalyhin) [inline authentication/LegacyAuthQuickstart/AuthenticationExample/AppDelegate.m:38]: @morganchen12 I'm not sure if the tags are affecting anything and if I updated them properly. Could you PTAL.

**Review by paulb777 — COMMENTED** [review]
Thanks Maksym! LGTM

Deferring review of doc tags and approval to @morganchen12

File: authentication/LegacyAuthQuickstart/AuthenticationExample/AppDelegate.m (hunk 1 above)
Comment (morganchen12) [reply authentication/LegacyAuthQuickstart/AuthenticationExample/AppDelegate.m:38]: Deleting this snippet is ok.

**Review by morganchen12 — APPROVED** [review]

File: authentication/LegacyAuthQuickstart/AuthenticationExample/AppDelegate.m (hunk 1 above)
Comment (maksymmalyhin) [reply authentication/LegacyAuthQuickstart/AuthenticationExample/AppDelegate.m:38]: SG. Please add me to reviewers for the updated docs to ensure they have correct snippets.

**Pull request merged**